### PR TITLE
feat : code update as per zaun

### PR DESCRIPTION
**Pull request by ocdbytes**
- Added makefile.
- Updated code as per new zaun implementation.
- removed hard coded artifacts which we can build.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -59,6 +65,12 @@ pub struct CliArgs {
...
 }
 
+#[derive(Serialize, Deserialize)]
+pub enum CoreContractMode {
+    Production,
+    Dev,
+}
+
 #[derive(Serialize, Deserialize)]
 pub struct ConfigFile {
```
Comment (apoorvsadana) [inline src/main.rs:75]: i don't know what's a good design to do this but looking at the config file, i don't know what optional things are needed for what commands. like `l1_eth_bridge_address` is needed for upgrades but not for others. we can add a TODO with this problem and solve it eventually

File: src/main.rs [hunk 2]
```diff
@@ -143,26 +175,39 @@ pub async fn main() {
...
             BootstrapperOutput { argent_setup_outputs: Some(output), ..Default::default() }
         }
         BootstrapMode::Braavos => {
-            let output = setup_braavos(None, &config_file, &clients).await;
+            let output = setup_braavos(
+                account,
+                &config_file,
+                &clients,
+                Felt::from_str(
+                    &config_file.udc_address.clone().expect("UDC Address not available in config. Run with SetupL2"),
```
Comment (apoorvsadana) [inline src/main.rs:200]:
```suggestion
                    &config_file.udc_address.clone().expect("UDC Address not available in config. Run with mode UDC"),
```

File: src/main.rs [hunk 3]
```diff
@@ -275,6 +320,43 @@ async fn setup_eth_bridge<'a>(
...
+) -> color_eyre::Result<()> {
+    let account = match account {
+        Some(account) => account,
+        None => get_account(clients, config_file).await,
+    };
+    upgrade_eth_token_to_cairo_1(
+        &account,
+        clients.provider_l2(),
+        Felt::from_str(
+            &config_file.l2_eth_token_proxy_address.clone().expect("l2_eth_token_proxy_address not in config."),
```
Comment (apoorvsadana) [inline src/main.rs:336]: do we want expects or should we throw errors? i think we had moved a lot of expects in the main file so i am in the camp of throwing errors and keeping it coherent

File: src/main.rs [hunk 4]
```diff
@@ -356,7 +439,7 @@ pub async fn setup_l2(config_file: &ConfigFile, clients: &Clients) -> Bootstrapp
     let account = get_account(clients, config_file).await;
 
     let core_contract_client = get_core_contract_client(config_file, clients);
-
+    println!(">>> get core contract client done");
```
Comment (apoorvsadana) [inline src/main.rs:442]:
```suggestion
```

File: src/main.rs [hunk 5]
```diff
@@ -372,7 +455,31 @@ pub async fn setup_l2(config_file: &ConfigFile, clients: &Clients) -> Bootstrapp
...
+    )
+    .await;
+    upgrade_eth_bridge_to_cairo_1(
+        &account,
+        clients.provider_l2(),
+        eth_bridge_setup_outputs.clone().l2_eth_bridge_proxy_address,
+        eth_bridge_setup_outputs.clone().l2_eth_proxy_address,
+    )
+    .await;
+    upgrade_l1_bridge(eth_bridge_setup_outputs.clone().l1_bridge_address, config_file).await.unwrap();
```
Comment (apoorvsadana) [inline src/main.rs:482]: let's call `upgrade_eth_bridge` directly?

File: src/setup_scripts/upgrade_eth_token.rs [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+use tokio::time::sleep;
+
+use crate::contract_clients::utils::{declare_contract, DeclarationInput, RpcAccount};
+use crate::helpers::account_actions::{get_contract_address_from_deploy_tx, AccountActions};
+use crate::utils::constants::{
+    EIC_ETH_TOKEN_CASM_PATH, EIC_ETH_TOKEN_SIERRA_PATH, NEW_ETH_TOKEN_CASM_PATH, NEW_ETH_TOKEN_SIERRA_PATH,
+};
+use crate::utils::wait_for_transaction;
+
+pub async fn upgrade_eth_token_to_cairo_1(
```
Comment (apoorvsadana) [inline src/setup_scripts/upgrade_eth_token.rs:16]: can you write a comment on top of all upgrade functions explaining what's exactly happening. maybe we can copy from the notion

File: src/tests/mod.rs [hunk 7]
```diff
@@ -150,6 +131,75 @@ async fn deposit_tests_both_bridges() -> Result<(), anyhow::Error> {
...
+        .arg("--rpc-methods")
+        .arg("unsafe")
+        .arg("--l1-endpoint")
+        .arg("http://localhost:8545")
+        .spawn()?;
+
+    env::set_current_dir("../").expect("Navigate back failed.");
+
+    // Madara build time (approx : 20 mins.)
+    sleep(Duration::from_secs(1200)).await;
```
Comment (apoorvsadana) [inline src/tests/mod.rs:192]: why can't we do `node.wait_for_ready().await;` like before

File: src/contract_clients/starknet_core_contract.rs [hunk 8]
```diff
@@ -15,22 +18,31 @@ use crate::contract_clients::core_contract::{
...
+impl CoreContractDeploy<StarknetCoreContract> for StarknetCoreContract {
     async fn deploy(clients: &Clients) -> Self {
-        let client = deploy_starknet_validity_behind_safe_proxy(clients.eth_client().signer().clone())
-            .await
-            .expect("Failed to deploy the starknet contact");
-
-        Self { core_contract_client: client }
+        let client = deploy_starknet_core_contract_behind_proxy(
+            clients.eth_client().signer().clone(),
+            ProxyVersion::SafeProxy3_0_2,
```
[the remark this answers is not in the document]
Comment (ocdbytes) [reply src/contract_clients/starknet_core_contract.rs:29]: Yes it mentions both the proxies. The thing is they both have different permissions and roles setup. In my opinion let's wait for their reply and let's see if our permissions are up to date in the current proxy setup.

File: src/contract_clients/starknet_dev_core_contract.rs [hunk 9]
```diff
@@ -15,22 +18,31 @@ use crate::contract_clients::core_contract::{
...
     async fn deploy(clients: &Clients) -> Self {
-        let client = deploy_starknet_sovereign_behind_unsafe_proxy(clients.eth_client().signer().clone())
-            .await
-            .expect("Failed to deploy the starknet contact");
-
-        Self { core_contract_client: client }
+        let client = deploy_starknet_core_contract_behind_proxy(
+            clients.eth_client().signer().clone(),
+            ProxyVersion::SafeProxy3_0_2,
+            CoreContractType::Dev,
```
[the remark this answers is not in the document]
Comment (ocdbytes) [reply src/contract_clients/starknet_dev_core_contract.rs:30]: This is specifically in starknet_dev_core_contract file. The decision is made in the main file whether the user has chosen the DEV or PRODUCTION contract type

File: src/main.rs (hunk 3 above)
Comment (ocdbytes) [reply src/main.rs:336]: let's create a separate issue for this ? or should I implement this in this PR itself ?

File: src/contract_clients/token_bridge.rs
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply src/contract_clients/token_bridge.rs:1]: can you create the issue

File: src/main.rs (hunk 3 above)
Comment (apoorvsadana) [reply src/main.rs:338]: i think this we should do in this PR

File: .github/workflows/linters-cargo.yml
[the remark this answers is not in the document]
Comment (ocdbytes) [reply .github/workflows/linters-cargo.yml:1]: Issue created : #22

**Review by Mohiiit — APPROVED** [review]

**Pull request merged**